app/src/main/java/org/simple/clinic/drugs/search/DrugSearchUpdate.kt

```diff
@@ -17,10 +16,19 @@ class DrugSearchUpdate : Update<DrugSearchModel, DrugSearchEvent, DrugSearchEffe
       is SearchQueryChanged -> searchDrugs(model, event)
       is DrugListItemClicked -> dispatch(OpenCustomDrugEntrySheetFromDrugList(event.drugId, event.patientUuid))
       is NewCustomDrugClicked -> dispatch(OpenCustomDrugEntrySheetFromDrugName(event.drugName, event.patientUuid))
-      is DrugFrequencyChoiceItemsLoaded -> noChange()
+      is DrugFrequencyChoiceItemsLoaded -> drugFrequencyChoiceItemsLoaded(model, event)
     }
   }
 
+  private fun drugFrequencyChoiceItemsLoaded(
+      model: DrugSearchModel,
+      event: DrugFrequencyChoiceItemsLoaded
+  ): Next<DrugSearchModel, DrugSearchEffect> {
+    val drugFrequencyToFrequencyChoiceItemMap = event.drugFrequencyChoiceItems.items.associateBy({ it.drugFrequency }, { it })
```

Comment:
I see we are converting the list to a map for the drug choice items, can't we just provide it as a map directly?

Reply:
Umm.. We are mapping it as DrugFrequency or MedicineFrequency to DrugFrequencyChoiceItem. Although you are right that approach sounds better and will remove a lot of code. We can merge this in for now. I will make that change in all the places and remove the unnecessary code.

Reply:
Yeah, sounds good. I am not saying we need to do this change here. I was just checking to see if there was any particular reason for going with this approach. We can also combine `DrugFrequency` & `MedicineFrequency` into one thing, right? Technically, I don't think there is any difference between them 🤔

Reply:
Yeah, we can take the whole DrugFrequency implementation out in a separate package and use DrugFrequency everywhere. Maintaining two different data classes is not necessary at all here.